prow/flagutil/kubernetes_cluster_clients.go

```diff
@@ -131,7 +140,9 @@ func (o *KubernetesOptions) AddKubeconfigChangeCallback(callback func()) error {
 
 // AddFlags injects Kubernetes options into the given FlagSet.
 func (o *KubernetesOptions) AddFlags(fs *flag.FlagSet) {
-	fs.StringVar(&o.kubeconfig, "kubeconfig", "", "Path to .kube/config file. If empty, uses the local cluster. All contexts other than the default are used as build clusters.")
+	fs.StringVar(&o.kubeconfig, "kubeconfig", "", "Path to .kube/config file. Mutually exclusive to --kubeconfig-dir. If both empty, uses the local cluster. All contexts other than the default are used as build clusters.")
```

Comment:
Why is it mutually exclusive when the `LoadCLusterConfig` method handles that case quite well?

Reply:
Also, what does "used as build clusters" mean in the context of a flagutil package? Is there any assumption built in that this is only used in the context of a system where "build clusters" are a thing?

Reply:
> Also, what does "used as build clusters" mean in the context of a flagutil package? Is there any assumption built in that this is only used in the context of a system where "build clusters" are a thing?

(The wording of "build clusters" was there for very long.)
I think `flagutil` is under `prow` and "build cluster" is a thing for `prow`.

Reply:
> Why is it mutually exclusive when the `LoadCLusterConfig` method handles that case quite well?

I will give it a try without "mutually exclusive".

Reply:
Updated.